### ISSUE TITLE
Kitizen not following shortest path

### ISSUE THREAD

**Issue by jaydenbascio** [candym0n/kitty-city#3]
# The problem
As a player, I would like my Kitizens to follow the shortest path to a workplace.
However, sometimes my Kitizens do not follow the shortest path, but instead get stuck in a triangle loop.
This makes it difficult to play the game as you do not know when your Kitizen will get stuck in a loop and be unable to continue working.

# Reproduction
1. Create 4 intersections
2. Connect each intersection to every other intersection
3. Create a house and a workplace
4. Connect a house to an intersection
5. Connect the workplace to a different intersection on the same edge
6. The cat should make it the first couple of times but get stuck in a triangle loop afterwards
![{858A05F2-B485-49FC-8AC0-05826971FA41}](https://github.com/user-attachments/assets/3ceb28b3-459e-4a65-a1ff-6af0b75278f2)

**Comment by candym0n** [comment]
Thank you for reporting this issue regarding Kitizen pathfinding in Kitty City. We appreciate your feedback and understand the frustration this may be causing.

While we acknowledge the importance of optimizing Kitizen paths, given the imminent deadline for Kitty City v2.0, our development team is currently prioritizing critical features and bug fixes. We apologize for any inconvenience this may cause.

As a temporary workaround, we recommend avoiding the creation of triangular or quadrilateral intersections in your city layouts. This will help to minimize the likelihood of Kitizens becoming trapped in loops.

We will continue to monitor this issue and explore potential solutions for future updates. Thank you for your understanding and patience.

**Comment by jaydenbascio** [comment]
Thank you for your insightful report concerning the suboptimal pathfinding exhibited by our Kitizens. We acknowledge the salience of this issue and comprehend the frustration it may engender.

While our development team is currently preoccupied with ensuring the timely release of Kitty City v2.0, we encourage you to leverage the collaborative spirit of our community to seek potential solutions. Engaging with other players or contributing to our open-source codebase may yield fruitful outcomes.

Should you possess the requisite technical acumen, we warmly invite you to submit a pull request to our GitHub repository. Your innovative approach to rectifying this anomaly could prove invaluable to our development endeavors.

We appreciate your understanding and patience as we strive to enhance the overall gaming experience.

**Comment by jaydenbascio** [comment]
Upon not being to fall asleep last night, knowing that 100% of my one-person market was unsatisfied with his experience here at Kitty City, I have given some thought as to what may be causing this problem.

One issue might be the road finding algorithm. The problem is that right now it ignores all roads that have already been found to prevent looping, but the very specific design that you setup would have only found a path with roads to the intersection at the bottom left, and completely disregarded that the road at the middle bottom could lead to the workplace.

This leads the cat into a triangle loop, as it only sees those three roads.

I have alerted my development team and we will get on this issue right away.

**Comment by jaydenbascio** [comment]
OK literally 3 minutes later I fixed the problem.

In roadProfit.js, I was passing alreadyFoundRoad into RoadProfit.FindBuildings, so other roads will ignore this road, even though we might need it later.

I fixed this by literally replacing alreadyFoundRoad with [...alreadyFoundRoad] to have the same information but the function will not alter alreadyFoundRoad directly.

This is very sad. It was literally just 5 characters. Well, this issue is closed now. It is fixed. All is good. I hope you are happy.

Fixed with commit d41570389ce31ec25eafbdd10180e6476e2723d7
